core/src/main/java/feign/ReflectiveFeign.java

```diff
@@ -47,18 +50,23 @@
   @Override
   public <T> T newInstance(Target<T> target) {
     Map<String, MethodHandler> nameToHandler = targetToHandlersByName.apply(target);
-    Map<Method, MethodHandler> methodToHandler = new LinkedHashMap<Method, MethodHandler>();
-    List<DefaultMethodHandler> defaultMethodHandlers = new LinkedList<DefaultMethodHandler>();
+    Map<Method, MethodHandler> methodToHandler = new LinkedHashMap<>();
+    List<DefaultMethodHandler> defaultMethodHandlers = new LinkedList<>();
 
     for (Method method : target.type().getMethods()) {
       if (method.getDeclaringClass() == Object.class) {
         continue;
-      } else if (Util.isDefault(method)) {
-        DefaultMethodHandler handler = new DefaultMethodHandler(method);
+      }
+      if (Util.isDefault(method)) {
+        DefaultMethodHandler handler = new DefaultMethodHandler(method, target, methodHandlerCustomizers);
         defaultMethodHandlers.add(handler);
         methodToHandler.put(method, handler);
       } else {
-        methodToHandler.put(method, nameToHandler.get(Feign.configKey(target.type(), method)));
+        MethodHandler handler = nameToHandler.get(Feign.configKey(target.type(), method));
+        for (MethodHandlerCustomizer customizer : methodHandlerCustomizers) {
+          handler = customizer.customize(target, method, handler);
+        }
```

Comment:
this for loop is repeated twice, I could add a static method `MethodHandlerCustomizer#customizeAll(List< MethodHandlerCustomizer>, Target, Method, MethodHandler)` to make it a one-liner.